content/docs/rfcs/36/README.md

```diff
@@ -0,0 +1,809 @@
+- `pubsubTopic`: The pubsub topic on which the message was received.
+- `messageId`: The message id.
+- `wakuMessage`: The message in [`JsonMessage`](#jsonmessage-type) format.
+
+### `extern void waku_set_event_callback(void* cb)`
+
+Register callback to act as event handler and receive application signals,
+which are used to react to asynchronous events in Waku.
+
+**Parameters**
+
+1. `void* cb`: callback that will be executed when an async event is emitted.
+  The function signature for the callback should be `void myCallback(char* jsonSignal)`
+
+## Node management
+
+### `JsonConfig` type
+
+Type holding a node configuration:
+
+```ts
+interface JsonSignal {
+    host?: string;
+    port?: number;
+    advertiseAddr?: string;
+    nodeKey?: string;
+    keepAliveInterval?: number;
+    relay?: boolean;
+    minPeersToPublish?: number
+}
+```
+
+Fields: 
+
+All fields are optional.
+If a key is `undefined`, or `null`, a default value will be set.
+
+- `host`: Listening IP address.
+  Default `0.0.0.0`.
+- `port`: Libp2p TCP listening port.
+  Default `60000`. 
+  Use `0` for random.
+- `advertiseAddr`: External address to advertise to other nodes.
+  Can be ip4, ip6 or dns4, dns6.
+  Default: ?
+- `nodeKey`: Secp256k1 private key in Hex format (`0x123...abc`).
+  Default random.
+- `keepAliveInterval`: Interval in seconds for pinging peers to keep the connection alive.
+  Default `20`.
+- `relay`: Enable relay protocol.
+  Default `true`.
+- `minPeersToPublish`: The minimum number of peers required on a topic to allow broadcasting a message.
+  Default `0`.
+
+For example:
+```json
+{
+  "host": "0.0.0.0",
+  "port": 60000,
+  "advertiseAddr": "1.2.3.4",
+  "nodeKey": "0x123...567",
+  "keepAliveInterval": 20,
+  "relay": true,
+  "minPeersToPublish": 0
+}
+```
+
+### `extern char* waku_new(char* jsonConfig)`
+
+Instantiates a Waku node.
+
+**Parameters**
+
+1. `char* jsonConfig`: JSON string containing the options used to initialize a go-waku node.
+   Type [`JsonConfig`](#jsonconfig-type).
+   It can be `NULL` to use defaults.
+
+**Returns**
+
+A [`JsonResponse`](#jsonresponse-type).
+If the execution is successful, the `result` field contains an `integer` which represents the `nodeId`.
+This node id MUST be used in following API calls to interact with the instantiated node.
+
+For example:
+
+```json
+{
+  "result": 0
+}
+```
+
+### `extern char* waku_start(int nodeId)`
+
+Start a Waku node mounting all the protocols that were enabled during the Waku node instantiation.
+
+**Parameters**
+
+1. `int nodeId`: The node identifier obtained from a successful execution of [`waku_new`](#extern-char-waku_newchar-jsonconfig).
+
+**Returns**
+
+A [`JsonResponse`](#jsonresponse-type).
+If the execution is successful, the `result` field is set to `null`.
+
+For example:
+
+```json
+{
+  "result": null
```

Comment:
@richard-ramos how do you process this kind of result?
In JS world it'd be awkward and I'd probably do

```js
if (!res.error) {
 // we're good
}
```

instead of 

```js
if (res.result === null) {
  // we're good
}
```

What about setting `true`:

```json
{
    "result": true
}
```

WDYT @richard-ramos @jm-clius

Reply:
I think `true` makes sense.

Reply:
I'd do `if (!res.error)`, but will change it to setting `result` to `true` so the response is consistent with the format used in other functions